### PR TITLE
System.cpp: Fix Create PPU Cache regression

### LINKED ISSUE #12412: [Regression]  Batch PPU creation fails
### Quick summary

Possibly same root cause as https://github.com/RPCS3/rpcs3/issues/12393
But firmware installs just fine for me

### Details

Anyways, Batch PPU creation regressed
Same as right click -> create PPU cache.
Booting the game like a normal person works fine.
I'm not a normal person

### Build with regression

v0.0.23-13834-fcd297ff

### Attach two log files

[Broken](https://github.com/RPCS3/rpcs3/files/9213451/RPCS3.log)
[Works](https://github.com/RPCS3/rpcs3/files/9213454/RPCS3.log)

### System configuration

My PC
https://pcpartpicker.com/list/

### PR DESCRIPTION
**Pull request by elad335**
Fix it by initializing things by the GUI thread as done in game-boot. And make it use native UI because why not. (rsx is initialized now)

Fixes #12412

### CODE REVIEW

File: rpcs3/Emu/system_progress.cpp [hunk 1]
```diff
@@ -82,7 +82,10 @@ void progress_dialog_server::operator()()
 			}
 		}
 
-		if (!skip_this_one && !native_dlg && (dlg = Emu.GetCallbacks().get_msg_dialog()))
+		if (!skip_this_one && !native_dlg && (Emu.BlockingCallFromMainThread([&dlg]()
```
Comment (Megamouse) [inline rpcs3/Emu/system_progress.cpp:85]: Just do this in get_msg_dialog. I don't understand why you do this in a super weird way here

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 2]
```diff
@@ -402,12 +402,30 @@ void gui_application::InitializeCallbacks()
...
-	callbacks.get_osk_dialog  = [this]() -> std::shared_ptr<OskDialogBase> { return m_show_gui ? std::make_shared<osk_dialog_frame>() : nullptr; };
+	callbacks.get_msg_dialog  = [this]() -> std::shared_ptr<MsgDialogBase> { return make_qt_object(this, static_cast<msg_dialog_frame*>(nullptr)); };
+	callbacks.get_osk_dialog  = [this]() -> std::shared_ptr<OskDialogBase> { return make_qt_object(this, static_cast<osk_dialog_frame*>(nullptr)); };
 	callbacks.get_save_dialog = []() -> std::unique_ptr<SaveDialogBase> { return std::make_unique<save_data_dialog>(); };
-	callbacks.get_sendmessage_dialog = [this]() -> std::shared_ptr<SendMessageDialogBase> { return std::make_shared<sendmessage_dialog_frame>(); };
-	callbacks.get_recvmessage_dialog = [this]() -> std::shared_ptr<RecvMessageDialogBase> { return std::make_shared<recvmessage_dialog_frame>(); };
+	callbacks.get_sendmessage_dialog = [this]() -> std::shared_ptr<SendMessageDialogBase> { return make_qt_object(this, static_cast<sendmessage_dialog_frame*>(nullptr)); };
+	callbacks.get_recvmessage_dialog = [this]() -> std::shared_ptr<RecvMessageDialogBase> { return make_qt_object(this, static_cast<recvmessage_dialog_frame*>(nullptr)); };
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:428]: these callbacks are already used in a blocking call.
So remove it in either place

**Comment by elad335** [comment]
@Darkhost1999 please submit an issue after merge.

**Comment by Darkhost1999** [comment]
Yea no problem. The error doesn't appear to be a blocker

**Pull request merged**